controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -173,7 +178,7 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	controllerutil.AddFinalizer(kcp, controlplanev1.KubeadmControlPlaneFinalizer)
 
 	// Make sure to reconcile the external infrastructure reference.
-	if err := r.reconcileExternalReference(ctx, cluster, kcp.Spec.InfrastructureTemplate); err != nil {
+	if err := r.reconcileExternalReference(ctx, cluster, kcp, kcp.Spec.InfrastructureTemplate); err != nil {
```

Comment:
I find this signature pretty weird, why not just pass in `kcp` and extract the template out in the method?